lib/image_render.dart

```diff
@@ -127,28 +131,33 @@ ImageRender networkImageRender({
         }
       });
 
-      image.image.resolve(ImageConfiguration()).addListener(
-            ImageStreamListener((ImageInfo image, bool synchronousCall) {
-              var myImage = image.image;
-              Size size = Size(myImage.width.toDouble(), myImage.height.toDouble());
-              if (!completer.isCompleted) {
-                completer.complete(size);
-              }
-            }, onError: (object, stacktrace) {
-              if (!completer.isCompleted) {
-                completer.completeError(object);
-              }
-            }),
-          );
+      var listener =
+          ImageStreamListener((ImageInfo image, bool synchronousCall) {
+        var myImage = image.image;
+        Size size = Size(myImage.width.toDouble(), myImage.height.toDouble());
+        if (!completer.isCompleted) {
+          completer.complete(size);
+        }
+      }, onError: (object, stacktrace) {
+        if (!completer.isCompleted) {
+          completer.completeError(object);
+        }
+      });
+
+      image.image.resolve(ImageConfiguration()).addListener(listener);
       return FutureBuilder<Size>(
         future: completer.future,
         builder: (BuildContext buildContext, AsyncSnapshot<Size> snapshot) {
+          if (completer.isCompleted) {
+            image.image.resolve(ImageConfiguration()).removeListener(listener);
+          }
```

Comment:
Ahh good thinking.... I was testing this yesterday and I couldn't figure it out, sorry for not responding. Thanks for the fix, much appreciated!!